group/assignment5/find_all_alphabets.py

```diff
@@ -0,0 +1,38 @@
+from collections import defaultdict
+from assignment5.graph import *
+from assignment5.find_alphabet import build_graph
+
+def find_all_alphabets(dictionary):
+    graph = build_graph(dictionary)
+    letters = graph.vertices
+    letters_to_add = list()
+    added_parents = defaultdict(int)
+    all_alphabets = list()
+    for l in letters:
+        if len(graph.parents[l]) == 0:
+            letters_to_add.append(l)
+    find_all_alphabets_with_prefix(graph, '', letters_to_add, added_parents,
+            all_alphabets)
+    return set(all_alphabets)
+
+
+def find_all_alphabets_with_prefix(graph, prefix, letters_to_add, added_parents,
+        all_alphabets):
+    if len(prefix) == len(graph.vertices):
+        all_alphabets.append(prefix)
+        return
+    if len(letters_to_add) == 0:
+        return
+    for l in letters_to_add:
```

Comment:
@aliskin  you can replace letters_to_add with queue collection and it will help you to avoid remove and copy method, but I think it is optional and maybe your implementation is also good)

Reply:
We cannot avoid copy and remove methods because we need a copy of this list for each recursive call. I don't see how queue collection could work in this case.